### PR TITLE
App-SRE Integration

### PR DESCRIPTION
**Pull request by ldelossa**
This PR will remain in draft until all RedHat App-SRE integrations are successfully working.

### CODE REVIEW

File: contrib/openshift/build_and_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    echo "could not set root to clair repository"
+    exit 1
+fi
+
+go mod vendor
+make container-build
+
+skopeo copy --dest-creds "${QUAY_USER}:${QUAY_TOKEN}" \
+    "docker-daemon:clair-local:latest" \
+    "docker://clairv4:latest"
```
Comment (ldelossa) [inline contrib/openshift/build_and_deploy.sh]: @tparikh fixed.

Comment (tparikh) [reply contrib/openshift/build_and_deploy.sh]: @ldelossa it needs to be `docker://quay.io/app-sre/clairv4` and not `docker://clairv4:latest` I think.

File: contrib/openshift/build_and_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+skopeo copy --dest-creds "${QUAY_USER}:${QUAY_TOKEN}" \
+    "docker-daemon:clair-local:latest" \
+    "docker://clairv4:${GIT_HASH}"
```
Comment (tparikh) [inline contrib/openshift/build_and_deploy.sh]: I don't see `GIT_HASH` set in the script. You can add GIT_HASH=`git rev-parse --short=7 HEAD` to the script. Note: ` around the command are not visible due to markdown formatting

**Review by tparikh — CHANGES_REQUESTED** [review]

File: contrib/openshift/build_and_deploy.sh (hunk 2 above)
Comment (ldelossa) [reply contrib/openshift/build_and_deploy.sh]: good catch! thought that was coming from the env that runs the script

**Comment by tparikh** [comment]
LGTM

**Review by tparikh — COMMENTED** [review]
@ldelossa Are Clair logs shipped to CloudWatch?

**Comment by ldelossa** [comment]
@tparikh not currently. I will be doing a full metrics pass on ClairV4 and ClairCore before we move further into production.

File: contrib/openshift/manifests/manifests.yaml [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    value: "4096Mi"
+    displayName: the matcher's memory requests in vCPUs
+  #
+  # shared params
+  #
+  - name: CLAIR_SECRET
+    value: "clair"
+    displayName: the vault secret containing clair's config
+  - name: CLAIR_IMAGE
+    value: ""
```
Comment (rrati) [inline contrib/openshift/manifests/manifests.yaml]: Wouldn't you want this to point to a default image?  Same for tag

Comment (ldelossa) [reply contrib/openshift/manifests/manifests.yaml]: Let me take a look, i was under the impression these values do not get used.

Comment (rrati) [reply contrib/openshift/manifests/manifests.yaml]: These should be default values.  Deploments can override, but it doesn't make sense imo for the image to be blank as a default.

Comment (ldelossa) [reply contrib/openshift/manifests/manifests.yaml]: Got it, I'll fix

**Comment by ldelossa** [comment]
@rrati updated the image tags.

**Review by hdonnay — APPROVED** [review]

**Pull request merged**